staging/src/k8s.io/apiserver/pkg/endpoints/metrics/metrics.go

```diff
@@ -172,6 +179,67 @@ var (
 		currentInflightRequests,
 		requestTerminationsTotal,
 	}
+
+	// these are the known (e.g. whitelisted/known) content types which we will report for
+	// request metrics. Any other RFC compliant content types will be aggregated under 'unknown'
+	knownMetricContentTypes = utilsets.NewString(
+		"application/apply-patch+yaml",
+		"application/font-ttf",
+		"application/font-woff",
+		"application/font-woff2",
+		"application/javascript",
+		"application/json",
+		"application/json-patch+json",
+		"application/merge-patch+json",
+		"application/octet-stream",
+		"application/pdf",
+		"application/strategic-merge-patch+json",
+		"application/vnd.google.protobuf",
+		"application/vnd.kubernetes.protobuf",
+		"application/vnd.ms-fontobject",
+		"application/vnd.openxmlformats-officedocument.spreadsheetml.sheet",
+		"application/vnd.schemaregistry.v1+json",
+		"application/x-font-woff",
+		"application/x-gzip",
+		"application/x-javascript",
+		"application/x-pem-file",
+		"application/xml",
+		"application/yaml",
+		"cluster",
+		"font/ttf",
+		"font/woff",
+		"font/woff2",
+		"image/gif",
+		"image/jpeg",
+		"image/png",
+		"image/svg+xml",
+		"image/vnd.microsoft.icon",
+		"image/x-icon",
+		"namespace",
+		"resource",
```

Comment:
where are "namespace", "resource", and "cluster" set as content-type headers?

Reply:
It doesn't have to be done in our code, it can be done arbitrarily by a client.

Reply:
but... are those valid content-types the API server handles?

Reply:
I expected this to be bounded to a very small number of things (like our json/yaml/proto types and accepted patch types)

Reply:
this is still outstanding... I would recommend limiting this to the known mime-types we accept/serve